### ISSUE TITLE
TypeError: Cannot read property 'replace' of null

### ISSUE THREAD

**Issue by FelisCatus** [FelisCatus/SwitchyOmega#1159]
```
ERROR: [1] Unhandled rejection:
 TypeError: Cannot read property 'replace' of null
    at Object.requestFromUrl (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_pac.min.js:9760:27)
    at chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/background.js:108:37
From previous event:
    at ChromeTabs.actionForUrl (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/background.js:106:26)
    at ChromeTabs.processTab (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5849:17)
    at ChromeTabs.onUpdated (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5821:17)
    at chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5802:26
    at Array.forEach (<anonymous>)
    at Object.callback (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5799:21)
    at safeCallbackApply (extensions::uncaught_exception_handler:27:15)
    at handleResponse (extensions::sendRequest:67:7)
```

**Comment by FelisCatus** [comment]
@olloff The error seems to be hinting about a particular Chrome Tab that contains a URL with no protocol (or the library failed to parse the URL). Can you shed some light on what that URL could be? Any weird URLs, special pages (like Chrome pages?) did you open at that time? Any idea on how to reproduce this?

**Comment by olloff** [comment]
I was struggling to find a page without https-redirection, so this could be one of SSL error pages, like https://mashable.com/, https://www.tastefullyoffensive.com/ or https://sunline.ua/
Log dumped from 2.5.0 on Chrome 60 (stable) over Windows 7 VM.

Honestly, I doubt you can reproduce this without efforts. The last 100 tries across browser history in same condition gave nothing, so either it's heisenbug, or I do something else that particular moment (changing system of switchy proxy settings, for example).
I'll keep trying, tho.

**Comment by FelisCatus** [comment]
Thanks for trying. I've seen errors like this from other users a few times so it makes sense to patch this even if I don't know the root cause. Please feel free to keep trying or stop it if it just takes too much effort.

**Comment by olloff** [comment]
I've got it, but I'm not proud at all.

This situation can occur when Chrome lacks cached or saved proxy credentials.

1. Delete proxy credentials from password manager.
2. Set up system proxy settings as a proxy with auth (i used to test it on HTTP and Secure settings).
3. Switch to System proxy profile.
4. Try to open a page.
5. Cancel auth popup.

Issue is repeatable and can occur on step 5 or step 4 (I can miss something, probably).
What's more important, I've forgotten to search for similar issues and there was at least one with all info needed: https://github.com/FelisCatus/SwitchyOmega/issues/1151

```
Options << Options#applyProfile [
    "ukraine-sanctions"
]
Options << Options#updateProfile [
    [
        "ukraine-sanctions",
        "direct",
        "el-ninio-proxy"
    ]
]
Options << Options#updateProfile []
ProxyAuth <ip:port> 0 undefined
ProxyAuth <ip:port> 0 undefined
Options << Options#applyProfile [
    "direct"
]
Options << Options#updateProfile [
    [
        "direct"
    ]
]
ERROR: [1] Unhandled rejection:
 TypeError: Cannot read property 'replace' of null
    at Object.requestFromUrl (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_pac.min.js:9760:27)
    at chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/background.js:108:37
From previous event:
    at ChromeTabs.actionForUrl (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/background.js:106:26)
    at ChromeTabs.processTab (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5849:17)
    at ChromeTabs.onUpdated (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5821:17)
    at chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5802:26
    at Array.forEach (<anonymous>)
    at Object.callback (chrome-extension://padekgcemlokbadohgkifijomclgjgif/js/omega_target_chromium_extension.min.js:5799:21)
    at safeCallbackApply (extensions::uncaught_exception_handler:27:15)
    at handleResponse (extensions::sendRequest:67:7)
```

**Comment by FelisCatus** [comment]
Got it.

**Comment by FelisCatus** [comment]
Can't reproduce this using polipo on Linux. I think this happened because polipo still shows you a page saying "407 Proxy authentication required" even if you cancel the dialog.

Is there any different about how your proxy server behave? Does it simply drop the connection?

**Comment by olloff** [comment]
> Does it simply drop the connection?

This is most likely reason, yeah. Can't tell if it normal to 3proxy or my instance is miconfigured. 

Anyway I've checked it again and spot another symptom: if I change the profile for 2-3 times, exact same error will be thrown to console every time and exact same error page will be shown in browser (i need to discard proxy auth popup every time to see error page). 
But after that, browser shows me an empty page instead of error page. A page without an address (not about:blank), where extension button hangs and its not possible to switch profile again on this particular page (other pages are working fine) unless to enter new address or to return to the previous page.
So, eventually, you can be right about 
> Chrome Tab that contains a URL with no protocol

**Comment by FelisCatus** [comment]
Patched. Let me know once you have any updates so that I can fix the root cause.

**Comment by FelisCatus** [comment]
Fixed on v2.5.1. Published on Chrome Web Store. Firefox version still under review.
